src/components/diff/DiffWidget.tsx

```diff
@@ -55,8 +62,27 @@ export function openDiffView(
     showDialog({
       title: 'Diff Not Supported',
       body: `Diff is not supported for ${PathExt.extname(
-        path
+        filePath
       ).toLocaleLowerCase()} files.`
     });
   }
 }
+
+/**
+ * Gets the path of the file relative to the Jupyter server root. This resolves the server root path after calling
+ * the `/git/server_root` REST API, and uses the Git repo root and the file path relative to the repo root to resolve
+ * the rest.
+ * @param path the file path relative to Git repo root
+ * @param topRepoPath the Git repo root path
+ */
+export async function getRelativeFilePath(
```

Comment:
How often is this function called? It seems like you should only ever need to call `/git/server_root` once per session

Reply:
This is called `once` when the user clicks the "Diff" button. This isn't invoked periodically on refresh or anything so its not a high call-rate use-case. 
I did think about calling it once during initialization but plumbing the value down through all the `Widgets` didn't seem worth the complexity in the current state. We can always refactor to make this more top-level if the need arises.

Reply:
Caching the response in a `static` attribute would be a nice solution.

Reply:
@fcollonval : Thats a good suggestion. I switched to just saving the result in a variable as a cache